### PR TITLE
[WFLY-5742] Fix test coverage for @RunAs in servlets

### PR DESCRIPTION
**Pull request by olukas**
JIRA: https://issues.jboss.org/browse/WFLY-5742

### CODE REVIEW

**Comment by bstansberry** [comment]
This doesn't belong in manual mode. The stopping the server business is redundant. Our CI runs will take forever if we allow a practice of adding manualmode tests to validate this kind of detail.

**Comment by olukas** [comment]
Test case was moved from manualmode to basic.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/runas/ServletRunAsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        deployer.deploy(DEPLOYMENT);
+
+        URL servletUrl = new URL(url.toExternalForm() + RunAsAdminServlet.SERVLET_PATH.substring(1) + "?"
+                + Utils.encodeQueryParam(CallProtectedEjbServlet.FILE_PARAM, CORRECT_ROLE_AND_STOP_SERVER.getAbsolutePath()));
+        Utils.makeCall(servletUrl.toURI(), HTTP_OK);
+        servletUrl = new URL(url.toExternalForm() + RunAsUserServlet.SERVLET_PATH.substring(1) + "?"
+                + Utils.encodeQueryParam(CallProtectedEjbServlet.FILE_PARAM, INCORRECT_ROLE_AND_STOP_SERVER.getAbsolutePath()));
+        Utils.makeCall(servletUrl.toURI(), HTTP_OK);
+
+        ServerReload.executeReloadAndWaitForCompletion(managementClient.getControllerClient(), 50000);
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/runas/ServletRunAsTestCase.java]: Why?

Is there something special about this scenario that makes it different from all the other undeployment work we do when stopping servers?

Comment (olukas) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/runas/ServletRunAsTestCase.java]: There was issue in Wildfly 10.0.0.CR1 when @RunAs for destroy() method works correctly for undeployment on running server but does not work during stopping server. It was fixed by PR https://github.com/wildfly/wildfly/pull/8241. For that reason this scenario is also covered in this test case.

You can simply reproduce it by running this test case with wildfly-10.0.0.CR1 server. Test scenario checkDestroyMethodInStopServerWithCorrectRole will fail due to unsuccessful EJB invocation in destroy() method of servlet.

**Pull request merged**